src/main/java/reactor/kafka/receiver/internals/ConsumerEventLoop.java

```diff
@@ -230,12 +236,22 @@ public void run() {
                     long r = requested;
                     if (r > 0) {
                         if (!awaitingTransaction.get()) {
-                            consumer.resume(consumer.assignment());
+                            if (pausedByUs.getAndSet(false)) {
+                                Set<TopicPartition> toResume = new HashSet<>(consumer.assignment());
+                                toResume.removeAll(this.pausedByUser);
+                                consumer.resume(toResume);
```

Comment:
is it relevant to keep the `pausedByUser` collection full after this point?
ie. all the branches where `pausedByUs == false` will clear the collection and refill it, but that clear is thus done in a lazy fashion

Reply:
Good point; yes, it can be cleared here.